crawler.py

```diff
@@ -369,7 +375,13 @@ def crawl(self):
         visited = []
         old_snitches = {}
 
+        excluded_tlds = self.exclude.split(",")
+
         for i, domain in enumerate(domains):
+            # checks to see if TLD of domain should be excluded from scan
+            extractedDomain = extract(domain)
+            if extractedDomain.suffix in excluded_tlds:
```

Comment:
We should probably move the act of excluding TLDs to the place where we get the list of domains. This is so we have the best shot at actually visiting `n_sites` domains.

For example, right now if you exclude "com" and run for ten domains, the crawler will end up visiting zero sites. What should happen is that we should visit the top ten non-.com domains on the list.

Reply:
didn't think of that, thank you! most recent commit now addresses this concern. What it doesn't handle is the rare occurrence that the returned list of domains could be shorter than `n_sites` ... It's so distant an edge case that I don't even know if it's worth handling -- what do you think?

Reply:
Why do you write this case isn't handled? The function does its best and returns as many domains up to `n_sites` as it can. If there are 10 domains total, `n_sites` is 10, `exclude` is `com`, and (let's say) only two domains are non-.com, two domains will be returned. What else can you do? I guess you can throw an exception, but why?

Reply:
That's what I was wondering -- if it's worth putting an exception in or not. Sounds like you have the same instinct I do; that it's not worth it, and best to leave the method as is where it returns as many domains up to `n_sites` as possible